src/components/MlGeoJsonLayer/MlGeoJsonLayer.tsx

```diff
@@ -168,7 +170,10 @@ const MlGeoJsonLayer = (props: MlGeoJsonLayerProps) => {
 	useLayer({
 		mapId: props.mapId,
 		options: {
-			source: props.labelProp ? 'source-' + layerId.current : undefined,
+			...(typeof props?.options?.source !== 'undefined' &&
```

Comment:
props.labelProp must still be checked and in case it is undefined options.source prop of label/symbol useLayer hook must be set to undefined to prevent its initialization.